elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -1269,50 +1239,41 @@ protected void addInverseRelation(String relationName, Object relationValue) {
             Class<A> annotationClass,
             A annotation,
             PersistentResource resource) {
-        if (resource.getRequestScope().getSecurityMode() == SecurityMode.BYPASS_SECURITY) {
-            return;
-        }
-        Class<? extends Check>[] anyChecks;
-        Class<? extends Check>[] allChecks;
-        try {
-            anyChecks = (Class<? extends Check>[]) annotationClass
-                    .getMethod("any").invoke(annotation, (Object[]) null);
-            allChecks = (Class<? extends Check>[]) annotationClass
-                    .getMethod("all").invoke(annotation, (Object[]) null);
-        } catch (ReflectiveOperationException e) {
-            throw new InvalidSyntaxException("Unknown permission " + annotationClass.getName(), e);
-        }
+        PermissionManager.ExtractedChecks checks = PermissionManager.extractChecks(annotationClass, annotation);
```

Comment:
Can we obliterate this function from this class now that we have a 'PermissionManager'?  It would be nice to have similar logic all in one place.